### ISSUE TITLE
Potential bug around passing incorrect hidden state to hydra value head 

### ISSUE THREAD

**Issue by jihan-yin** [CarperAI/trlx#341]
### 🐛 Describe the bug

For both the seq2seq and causal models, the implementation for the hydra head seems to be incorrect.

We are grabbing the very last hidden state from the decoder and passing that to the hydra head. [Code](https://github.com/CarperAI/trlx/blob/main/trlx/models/modeling_ppo.py#L871)

However, when we have right padding on the outputs, which is done [here](https://github.com/CarperAI/trlx/blob/main/trlx/trainer/accelerate_ppo_trainer.py#L337), we end up grabbing the wrong index of the sequence for all of the outputs. I believe we should instead be grabbing the index of the hidden state based on where the padding starts, like how we do it for the reward model implementation [here](https://github.com/CarperAI/trlx/blob/main/examples/summarize_rlhf/reward_model/reward_model.py#L71).

I did a quick basic test locally with FLAN-T5 to verify that the hidden states are different when we index from right padding tokens, even when they are masked (though we don't do masking in the TRLX code here).

```
from transformers import AutoTokenizer, AutoModel
tokenizer = AutoTokenizer.from_pretrained('google/flan-t5-xxl')
model = AutoModel.from_pretrained('google/flan-t5-xl', device_map='auto')

encoder_prompts = ['Which way did the dog run?', 'How big was he?']
decoder_prompts = ['<pad>That way!', '<pad>He was really, really big. Bigger than you can imagine!']
encoder_inputs = tokenizer(encoder_prompts, return_tensors='pt', padding=True)
decoder_inputs = tokenizer(decoder_prompts, return_tensors='pt', padding=True)

outputs = model(input_ids = encoder_inputs['input_ids'],
                attention_mask = encoder_inputs['attention_mask'],
                decoder_input_ids = decoder_inputs['input_ids'],
                decoder_attention_mask = decoder_inputs['attention_mask'],
                output_hidden_states=True,
                return_dict=True)
print(outputs.decoder_hidden_states[-1][0][-1])
print(outputs.decoder_hidden_states[-1][0][-2])
```

**Comment by jihan-yin** [comment]
Ah I misunderstood. We run the value head over the entire sequence and only slice the relevant values
